### PR TITLE
Qt/GraphicsWindow: Disable unsupported options

### PR DESCRIPTION
**Pull request by spycrab**
Fixes [issue #11016](https://bugs.dolphin-emu.org/issues/11016).

### CODE REVIEW

File: Source/Core/DolphinQt2/Config/Graphics/HacksWidget.cpp [hunk 1]
```diff
@@ -98,6 +101,12 @@ void HacksWidget::CreateWidgets()
   setLayout(main_layout);
 }
 
+void HacksWidget::OnBackendChanged()
+{
+  m_gpu_texture_decoding->setEnabled(g_Config.backend_info.bSupportsGPUTextureDecoding);
```
Comment (riking) [inline Source/Core/DolphinQt2/Config/Graphics/HacksWidget.cpp]: These should also be done when the widgets are created.

Comment (spycrab) [reply Source/Core/DolphinQt2/Config/Graphics/HacksWidget.cpp]: It is now.

**Comment by leoetlino** [comment]
Is it better to disable unsupported options or completely hide them? Right now, WX does the latter but I'm not sure it's the best approach.

**Comment by MayImilae** [comment]
I think disabled is better, honestly. It's kind of disconcerting when options just disappear, not to mention having it disabled gives a nudge to someone *cough stenzek cough* to implement the feature. :P 

Also, maybe have some sort of pop up tooltip explaining that it's disabled on this backend?

**Comment by JMC47** [comment]
Works fine for me.  I like it better like this.  Popup text takes a while but is consistent with other applications.

File: Source/Core/DolphinQt2/Config/Graphics/EnhancementsWidget.cpp [hunk 2]
```diff
@@ -165,6 +165,14 @@ void EnhancementsWidget::LoadSettings()
       m_pp_effect->setCurrentIndex(m_pp_effect->count() - 1);
   }
 
+  const bool supports_postprocessing = g_Config.backend_info.bSupportsPostProcessing;
+  m_pp_effect->setEnabled(supports_postprocessing);
+
+  if (!supports_postprocessing)
```
Comment (leoetlino) [inline Source/Core/DolphinQt2/Config/Graphics/EnhancementsWidget.cpp]: This if statement should be braced as it's more than one line long.

Comment (spycrab) [reply Source/Core/DolphinQt2/Config/Graphics/EnhancementsWidget.cpp]: Done.

**Pull request merged**